> > Doesn't build on darwin:
> 
> It's building on my machine.
Tried a different machine where I don't have xcode installed and i did indeed get the same error.

Removing `lib.optionalString stdenv.isLinux` from the preBuild gets it moving forward before i get this error. 
```
Undefined symbols for architecture x86_64:
  "std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::compare(unsigned long, unsigned long, char const*, unsigned long) const", referenced from:
      CEXEBuild::SetManifest() in build.o
  "std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >::find(wchar_t, unsigned long) const", referenced from:
      CEXEBuild::do_add_file_create_dir(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, int) in script.o
  "std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >::compare(unsigned long, unsigned long, wchar_t const*, unsigned long) const", referenced from:
      wmain(int, wchar_t**) in makenssi.o
      CEXEBuild::doCommand(int, LineParser&) in script.o
      CEXEBuild::do_add_file(wchar_t const*, int, int, int*, wchar_t const*, int, int*, std::__1::set<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::less<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > >, std::__1::allocator<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > > > const&, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, bool) in script.o
      CEXEBuild::pp_include(LineParser&) in scriptpp.o
  "std::__1::__vector_base_common<true>::__throw_length_error() const", referenced from:
      CEXEBuild::~CEXEBuild() in build.o
      CEXEBuild::write_output() in build.o
      CEXEBuild::uninstall_generate() in build.o
      void std::__1::vector<DialogItemTemplate*, std::__1::allocator<DialogItemTemplate*> >::__push_back_slow_path<DialogItemTemplate* const&>(DialogItemTemplate* const&) in DialogTemplate.o
      load_icon_res(CResourceEditor*, unsigned short) in icon.o
      load_icon_file(wchar_t const*) in icon.o
      set_main_icon(CResourceEditor*, unsigned short, std::__1::vector<Icon, std::__1::allocator<Icon> >, std::__1::vector<Icon, std::__1::allocator<Icon> >) in icon.o
      ...
  "std::__1::__basic_string_common<true>::__throw_length_error() const", referenced from:
      posix_dir_reader::read(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&) in dirreader.o
      Plugins::GetExports(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, bool, bool) in Plugins.o
  "std::logic_error::logic_error(char const*)", referenced from:
      CEXEBuild::init_shellconstantvalues() in build.o
      void std::__1::vector<DialogItemTemplate*, std::__1::allocator<DialogItemTemplate*> >::__push_back_slow_path<DialogItemTemplate* const&>(DialogItemTemplate* const&) in DialogTemplate.o
      void std::__1::vector<IconPair, std::__1::allocator<IconPair> >::__push_back_slow_path<IconPair const&>(IconPair const&) in icon.o
      std::__1::vector<CResourceDirectoryEntry*, std::__1::allocator<CResourceDirectoryEntry*> >::insert(std::__1::__wrap_iter<CResourceDirectoryEntry* const*>, CResourceDirectoryEntry* const&) in ResourceEditor.o
      std::__1::deque<CResourceDirectory*, std::__1::allocator<CResourceDirectory*> >::__add_back_capacity() in ResourceEditor.o
      std::__1::__split_buffer<CResourceDirectory**, std::__1::allocator<CResourceDirectory**> >::push_back(CResourceDirectory**&&) in ResourceEditor.o
      std::__1::__split_buffer<CResourceDirectory**, std::__1::allocator<CResourceDirectory**> >::push_front(CResourceDirectory**&&) in ResourceEditor.o
      ...
  "std::length_error::~length_error()", referenced from:
      void std::__1::vector<DialogItemTemplate*, std::__1::allocator<DialogItemTemplate*> >::__push_back_slow_path<DialogItemTemplate* const&>(DialogItemTemplate* const&) in DialogTemplate.o
      void std::__1::vector<IconPair, std::__1::allocator<IconPair> >::__push_back_slow_path<IconPair const&>(IconPair const&) in icon.o
      std::__1::vector<CResourceDirectoryEntry*, std::__1::allocator<CResourceDirectoryEntry*> >::insert(std::__1::__wrap_iter<CResourceDirectoryEntry* const*>, CResourceDirectoryEntry* const&) in ResourceEditor.o
      std::__1::deque<CResourceDirectory*, std::__1::allocator<CResourceDirectory*> >::__add_back_capacity() in ResourceEditor.o
      std::__1::__split_buffer<CResourceDirectory**, std::__1::allocator<CResourceDirectory**> >::push_back(CResourceDirectory**&&) in ResourceEditor.o
      std::__1::__split_buffer<CResourceDirectory**, std::__1::allocator<CResourceDirectory**> >::push_front(CResourceDirectory**&&) in ResourceEditor.o
      std::__1::__split_buffer<CResourceDirectory**, std::__1::allocator<CResourceDirectory**>&>::push_back(CResourceDirectory**&&) in ResourceEditor.o
      ...
  "std::out_of_range::~out_of_range()", referenced from:
      CEXEBuild::init_shellconstantvalues() in build.o
  "std::runtime_error::runtime_error(char const*)", referenced from:
      CEXEBuild::initialize(wchar_t const*) in build.o
      CEXEBuild::SetManifest() in build.o
      load_icon_res(CResourceEditor*, unsigned short) in icon.o
      load_icon_file(wchar_t const*) in icon.o
      generate_unicons_offsets(unsigned char*, unsigned long, unsigned char*, unsigned short) in icon.o
      CResourceEditor::GetNTHeaders(unsigned char*) in ResourceEditor.o
      CResourceEditor::GetResourceDirectory(unsigned char*, unsigned int, _IMAGE_NT_HEADERS32*, unsigned int*, unsigned int*) in ResourceEditor.o
      ...
  "std::runtime_error::~runtime_error()", referenced from:
      CEXEBuild::initialize(wchar_t const*) in build.o
      CEXEBuild::SetManifest() in build.o
      load_icon_res(CResourceEditor*, unsigned short) in icon.o
      load_icon_file(wchar_t const*) in icon.o
      generate_unicons_offsets(unsigned char*, unsigned long, unsigned char*, unsigned short) in icon.o
      CResourceEditor::GetNTHeaders(unsigned char*) in ResourceEditor.o
      CResourceEditor::GetResourceDirectory(unsigned char*, unsigned int, _IMAGE_NT_HEADERS32*, unsigned int*, unsigned int*) in ResourceEditor.o
      ...
  "std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::append(char const*)", referenced from:
      posix_dir_reader::read(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&) in dirreader.o
      manifest::addappendstring(wchar_t const*, wchar_t const*) in manifest.o
      manifest::generate(manifest::comctl, manifest::exec_level, manifest::SPECIFICATION const&) in manifest.o
  "std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::append(char const*, unsigned long)", referenced from:
      manifest::generate(manifest::comctl, manifest::exec_level, manifest::SPECIFICATION const&) in manifest.o
  "std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::assign(char const*)", referenced from:
      manifest::generate(manifest::comctl, manifest::exec_level, manifest::SPECIFICATION const&) in manifest.o
  "std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::basic_string(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
      manifest::addappendstring(wchar_t const*, wchar_t const*) in manifest.o
      void std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >::__push_back_slow_path<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in manifest.o
  "std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >::__init(wchar_t const*, unsigned long)", referenced from:
      CEXEBuild::initialize(wchar_t const*) in build.o
      CEXEBuild::set_target_architecture_data() in build.o
      CEXEBuild::pack_exe_header() in build.o
      CEXEBuild::write_output() in build.o
      CEXEBuild::initialize_default_plugins(bool) in build.o
      CEXEBuild::print_bad_targettype_parameter(wchar_t const*, wchar_t const*) const in build.o
      dir_reader::dir_reader() in dirreader.o
      ...
  "std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >::__init(wchar_t const*, unsigned long, unsigned long)", referenced from:
      std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > std::__1::operator+<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, wchar_t const*) in build.o
      Plugins::AddPluginsDir(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, bool, bool) in Plugins.o
      std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > std::__1::operator+<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, wchar_t const*) in Plugins.o
      CEXEBuild::do_add_file(wchar_t const*, int, int, int*, wchar_t const*, int, int*, std::__1::set<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::less<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > >, std::__1::allocator<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > > > const&, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, bool) in script.o
      CEXEBuild::add_file(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, int, wchar_t const*, int, int*) in script.o
      std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > std::__1::operator+<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, wchar_t const*) in script.o
      std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > std::__1::operator+<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >(wchar_t const*, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&) in script.o
      ...
  "std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >::append(wchar_t const*)", referenced from:
      CEXEBuild::initialize(wchar_t const*) in build.o
      CEXEBuild::set_compressor(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, bool) in build.o
      CEXEBuild::initialize_default_plugins(bool) in build.o
      CEXEBuild::load_stub() in build.o
      CEXEBuild::print_bad_targettype_parameter(wchar_t const*, wchar_t const*) const in build.o
      wmain(int, wchar_t**) in makenssi.o
      CEXEBuild::doCommand(int, LineParser&) in script.o
      ...
  "std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >::append(wchar_t const*, unsigned long)", referenced from:
      CEXEBuild::set_compressor(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, bool) in build.o
      std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > std::__1::operator+<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, wchar_t const*) in build.o
      Plugins::AddPluginsDir(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, bool, bool) in Plugins.o
      Plugins::GetExports(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, bool, bool) in Plugins.o
      std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > std::__1::operator+<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, wchar_t const*) in Plugins.o
      CEXEBuild::doCommand(int, LineParser&) in script.o
      CEXEBuild::do_add_file(wchar_t const*, int, int, int*, wchar_t const*, int, int*, std::__1::set<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::less<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > >, std::__1::allocator<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > > > const&, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, bool) in script.o
      ...
  "std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >::assign(wchar_t const*)", referenced from:
      CEXEBuild::initialize(wchar_t const*) in build.o
      CEXEBuild::resolve_coderefs(wchar_t const*) in build.o
      posix_dir_reader::read(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&) in dirreader.o
      wmain(int, wchar_t**) in makenssi.o
      CEXEBuild::doParse(wchar_t const*) in script.o
      CEXEBuild::doCommand(int, LineParser&) in script.o
      CEXEBuild::do_add_file(wchar_t const*, int, int, int*, wchar_t const*, int, int*, std::__1::set<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::less<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > >, std::__1::allocator<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > > > const&, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, bool) in script.o
      ...
  "std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >::insert(unsigned long, wchar_t const*)", referenced from:
      CEXEBuild::do_add_file_create_dir(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, int) in script.o
  "std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >::replace(unsigned long, unsigned long, wchar_t const*)", referenced from:
      replace_all(wchar_t const*, wchar_t const*, wchar_t const*) in util.o
  "std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >::push_back(wchar_t)", referenced from:
      Plugins::AddPluginsDir(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, bool, bool) in Plugins.o
      CEXEBuild::doCommand(int, LineParser&) in script.o
      CEXEBuild::do_add_file(wchar_t const*, int, int, int*, wchar_t const*, int, int*, std::__1::set<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::less<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > >, std::__1::allocator<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > > > const&, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, bool) in script.o
      CEXEBuild::add_file(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, int, wchar_t const*, int, int*) in script.o
      make_friendly_resource_path(wchar_t const*, wchar_t const*, unsigned short) in util.o
  "std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >::basic_string(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&)", referenced from:
      CEXEBuild::initialize(wchar_t const*) in build.o
      std::__1::pair<std::__1::__tree_iterator<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::__tree_node<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, void*>*, long>, bool> std::__1::__tree<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::less<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > >, std::__1::allocator<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > > >::__emplace_unique_key_args<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&>(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&) in dirreader.o
      Plugins::GetExports(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, bool, bool) in Plugins.o
      Plugins::FindDllPath(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >&) in Plugins.o
      Plugins::GetCommandInfo(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >&, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >&) in Plugins.o
      std::__1::pair<std::__1::__tree_iterator<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::__tree_node<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, void*>*, long>, bool> std::__1::__tree<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, STL::string_nocasecmpless<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, wchar_t>, std::__1::allocator<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > > >::__emplace_unique_key_args<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&>(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&) in Plugins.o
      std::__1::pair<std::__1::__tree_iterator<std::__1::__value_type<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > >, std::__1::__tree_node<std::__1::__value_type<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > >, void*>*, long>, bool> std::__1::__tree<std::__1::__value_type<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > >, std::__1::__map_value_compare<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::__value_type<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > >, STL::string_nocasecmpless<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, wchar_t>, true>, std::__1::allocator<std::__1::__value_type<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > > > >::__emplace_unique_key_args<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::piecewise_construct_t const&, std::__1::tuple<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&>, std::__1::tuple<> >(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, std::__1::piecewise_construct_t const&, std::__1::tuple<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&>&&, std::__1::tuple<>&&) in Plugins.o
      ...
  "std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >::basic_string(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, unsigned long, unsigned long, std::__1::allocator<wchar_t> const&)", referenced from:
      void PrintPluginDirsHelper::print<std::__1::map<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, STL::string_nocasecmpless<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, wchar_t>, std::__1::allocator<std::__1::pair<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > > > > >(std::__1::map<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, STL::string_nocasecmpless<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, wchar_t>, std::__1::allocator<std::__1::pair<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > > > > const&, char const*) in Plugins.o
      CEXEBuild::do_add_file(wchar_t const*, int, int, int*, wchar_t const*, int, int*, std::__1::set<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::less<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > >, std::__1::allocator<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > > > const&, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, bool) in script.o
      get_string_prefix(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&) in util.o
      get_string_suffix(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&) in util.o
  "std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >::operator=(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&)", referenced from:
      CEXEBuild::initialize(wchar_t const*) in build.o
      Plugins::GetExports(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, bool, bool) in Plugins.o
      CEXEBuild::do_add_file(wchar_t const*, int, int, int*, wchar_t const*, int, int*, std::__1::set<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >, std::__1::less<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > >, std::__1::allocator<std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > > > const&, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, bool) in script.o
      CEXEBuild::do_add_file_create_dir(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, int) in script.o
      CEXEBuild::pp_include(LineParser&) in scriptpp.o
  "std::bad_alloc::bad_alloc()", referenced from:
      CEXEBuild::ERROR_MSG(wchar_t const*, ...) const in build.o
      CEXEBuild::add_string(wchar_t const*, int, unsigned int) in build.o
      CEXEBuild::warninghelper(DIAGCODE, bool, wchar_t const*, __va_list_tag*) in build.o
      ExpandoString<wchar_t, 214ul>::StrFmt(wchar_t const*, ...) in build.o
      CEXEBuild::doParse(int, wchar_t const*, ...) in script.o
      ExeHeadStringList::add(wchar_t const*, unsigned short, bool) in strlist.o
      operator new(unsigned long) in util.o
      ...
  "std::bad_alloc::~bad_alloc()", referenced from:
      CEXEBuild::ERROR_MSG(wchar_t const*, ...) const in build.o
      CEXEBuild::add_string(wchar_t const*, int, unsigned int) in build.o
      CEXEBuild::warninghelper(DIAGCODE, bool, wchar_t const*, __va_list_tag*) in build.o
      ExpandoString<wchar_t, 214ul>::StrFmt(wchar_t const*, ...) in build.o
      CEXEBuild::doParse(int, wchar_t const*, ...) in script.o
      ExeHeadStringList::add(wchar_t const*, unsigned short, bool) in strlist.o
      operator new(unsigned long) in util.o
      ...
  "std::terminate()", referenced from:
      ___clang_call_terminate in build.o
      ___clang_call_terminate in clzma.o
      ___clang_call_terminate in icon.o
      ___clang_call_terminate in lang.o
      ___clang_call_terminate in makenssi.o
      ___clang_call_terminate in mmap.o
      ___clang_call_terminate in Plugins.o
      ...
  "typeinfo for std::length_error", referenced from:
      void std::__1::vector<DialogItemTemplate*, std::__1::allocator<DialogItemTemplate*> >::__push_back_slow_path<DialogItemTemplate* const&>(DialogItemTemplate* const&) in DialogTemplate.o
      void std::__1::vector<IconPair, std::__1::allocator<IconPair> >::__push_back_slow_path<IconPair const&>(IconPair const&) in icon.o
      std::__1::vector<CResourceDirectoryEntry*, std::__1::allocator<CResourceDirectoryEntry*> >::insert(std::__1::__wrap_iter<CResourceDirectoryEntry* const*>, CResourceDirectoryEntry* const&) in ResourceEditor.o
      std::__1::deque<CResourceDirectory*, std::__1::allocator<CResourceDirectory*> >::__add_back_capacity() in ResourceEditor.o
      std::__1::__split_buffer<CResourceDirectory**, std::__1::allocator<CResourceDirectory**> >::push_back(CResourceDirectory**&&) in ResourceEditor.o
      std::__1::__split_buffer<CResourceDirectory**, std::__1::allocator<CResourceDirectory**> >::push_front(CResourceDirectory**&&) in ResourceEditor.o
      std::__1::__split_buffer<CResourceDirectory**, std::__1::allocator<CResourceDirectory**>&>::push_back(CResourceDirectory**&&) in ResourceEditor.o
      ...
  "typeinfo for std::out_of_range", referenced from:
      CEXEBuild::init_shellconstantvalues() in build.o
  "typeinfo for std::runtime_error", referenced from:
      CEXEBuild::initialize(wchar_t const*) in build.o
      CEXEBuild::SetManifest() in build.o
      GCC_except_table65 in build.o
      load_icon_res(CResourceEditor*, unsigned short) in icon.o
      load_icon_file(wchar_t const*) in icon.o
      generate_unicons_offsets(unsigned char*, unsigned long, unsigned char*, unsigned short) in icon.o
      GCC_except_table1 in Plugins.o
      ...
  "typeinfo for std::bad_alloc", referenced from:
      CEXEBuild::ERROR_MSG(wchar_t const*, ...) const in build.o
      CEXEBuild::add_string(wchar_t const*, int, unsigned int) in build.o
      CEXEBuild::warninghelper(DIAGCODE, bool, wchar_t const*, __va_list_tag*) in build.o
      ExpandoString<wchar_t, 214ul>::StrFmt(wchar_t const*, ...) in build.o
      CEXEBuild::doParse(int, wchar_t const*, ...) in script.o
      ExeHeadStringList::add(wchar_t const*, unsigned short, bool) in strlist.o
      operator new(unsigned long) in util.o
      ...
  "typeinfo for std::exception", referenced from:
      GCC_except_table52 in build.o
      GCC_except_table54 in build.o
      GCC_except_table62 in build.o
      GCC_except_table64 in build.o
      GCC_except_table71 in build.o
      GCC_except_table7 in icon.o
      GCC_except_table30 in lang.o
      ...
  "vtable for __cxxabiv1::__class_type_info", referenced from:
      typeinfo for SortedStringListND<string_t> in build.o
      typeinfo for SortedStringListND<constantstring> in build.o
      typeinfo for SortedStringListND<uservarstring> in build.o
      typeinfo for SortedStringListND<langstring> in build.o
      typeinfo for SortedStringList<define> in build.o
      typeinfo for ICompressor in build.o
      typeinfo for writer in build.o
      ...
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for __cxxabiv1::__si_class_type_info", referenced from:
      typeinfo for LangStringList in build.o
      typeinfo for TinyGrowBuf in build.o
      typeinfo for UserVarsStringList in build.o
      typeinfo for CZlib in build.o
      typeinfo for CBzip2 in build.o
      typeinfo for FastStringList in build.o
      typeinfo for page_writer in build.o
      ...
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for __cxxabiv1::__vmi_class_type_info", referenced from:
      typeinfo for CLZMA in clzma.o
      typeinfo for MMapBuf in mmap.o
      typeinfo for NCompress::NLZMA::CEncoder in LZMAEncoder.o
      typeinfo for NBT2::CMatchFinder in LZMAEncoder.o
      typeinfo for NBT3::CMatchFinder in LZMAEncoder.o
      typeinfo for NBT4::CMatchFinder in LZMAEncoder.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for std::length_error", referenced from:
      void std::__1::vector<DialogItemTemplate*, std::__1::allocator<DialogItemTemplate*> >::__push_back_slow_path<DialogItemTemplate* const&>(DialogItemTemplate* const&) in DialogTemplate.o
      void std::__1::vector<IconPair, std::__1::allocator<IconPair> >::__push_back_slow_path<IconPair const&>(IconPair const&) in icon.o
      std::__1::vector<CResourceDirectoryEntry*, std::__1::allocator<CResourceDirectoryEntry*> >::insert(std::__1::__wrap_iter<CResourceDirectoryEntry* const*>, CResourceDirectoryEntry* const&) in ResourceEditor.o
      std::__1::deque<CResourceDirectory*, std::__1::allocator<CResourceDirectory*> >::__add_back_capacity() in ResourceEditor.o
      std::__1::__split_buffer<CResourceDirectory**, std::__1::allocator<CResourceDirectory**> >::push_back(CResourceDirectory**&&) in ResourceEditor.o
      std::__1::__split_buffer<CResourceDirectory**, std::__1::allocator<CResourceDirectory**> >::push_front(CResourceDirectory**&&) in ResourceEditor.o
      std::__1::__split_buffer<CResourceDirectory**, std::__1::allocator<CResourceDirectory**>&>::push_back(CResourceDirectory**&&) in ResourceEditor.o
      ...
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for std::out_of_range", referenced from:
      CEXEBuild::init_shellconstantvalues() in build.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "___cxa_allocate_exception", referenced from:
      CEXEBuild::initialize(wchar_t const*) in build.o
      CEXEBuild::init_shellconstantvalues() in build.o
      CEXEBuild::ERROR_MSG(wchar_t const*, ...) const in build.o
      CEXEBuild::add_string(wchar_t const*, int, unsigned int) in build.o
      CEXEBuild::SetManifest() in build.o
      CEXEBuild::warninghelper(DIAGCODE, bool, wchar_t const*, __va_list_tag*) in build.o
      ExpandoString<wchar_t, 214ul>::StrFmt(wchar_t const*, ...) in build.o
      ...
  "___cxa_begin_catch", referenced from:
      ___clang_call_terminate in build.o
      CEXEBuild::AddVersionInfo() in build.o
      CEXEBuild::ProcessPages() in build.o
      CEXEBuild::SetVarsSection() in build.o
      CEXEBuild::SetManifest() in build.o
      CEXEBuild::UpdatePEHeader() in build.o
      CEXEBuild::write_output() in build.o
      ...
  "___cxa_end_catch", referenced from:
      CEXEBuild::AddVersionInfo() in build.o
      CEXEBuild::ProcessPages() in build.o
      CEXEBuild::SetVarsSection() in build.o
      CEXEBuild::SetManifest() in build.o
      CEXEBuild::UpdatePEHeader() in build.o
      CEXEBuild::write_output() in build.o
      CLZMA::CompressReal() in clzma.o
      ...
  "___cxa_free_exception", referenced from:
      CEXEBuild::initialize(wchar_t const*) in build.o
      CEXEBuild::init_shellconstantvalues() in build.o
      CEXEBuild::SetManifest() in build.o
      void std::__1::vector<DialogItemTemplate*, std::__1::allocator<DialogItemTemplate*> >::__push_back_slow_path<DialogItemTemplate* const&>(DialogItemTemplate* const&) in DialogTemplate.o
      load_icon_res(CResourceEditor*, unsigned short) in icon.o
      load_icon_file(wchar_t const*) in icon.o
      generate_unicons_offsets(unsigned char*, unsigned long, unsigned char*, unsigned short) in icon.o
      ...
  "___cxa_guard_acquire", referenced from:
      CDialogTemplate::Save(unsigned int&) in DialogTemplate.o
  "___cxa_guard_release", referenced from:
      CDialogTemplate::Save(unsigned int&) in DialogTemplate.o
  "___cxa_pure_virtual", referenced from:
      vtable for dir_reader in dirreader.o
      vtable for writer_sink in writer.o
  "___cxa_throw", referenced from:
      CEXEBuild::initialize(wchar_t const*) in build.o
      CEXEBuild::init_shellconstantvalues() in build.o
      CEXEBuild::ERROR_MSG(wchar_t const*, ...) const in build.o
      CEXEBuild::add_string(wchar_t const*, int, unsigned int) in build.o
      CEXEBuild::SetManifest() in build.o
      CEXEBuild::warninghelper(DIAGCODE, bool, wchar_t const*, __va_list_tag*) in build.o
      ExpandoString<wchar_t, 214ul>::StrFmt(wchar_t const*, ...) in build.o
      ...
  "___gxx_personality_v0", referenced from:
      CEXEBuild::~CEXEBuild() in build.o
      CEXEBuild::CEXEBuild(signed char, bool) in build.o
      CEXEBuild::initialize(wchar_t const*) in build.o
      CEXEBuild::set_compressor(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, bool) in build.o
      CEXEBuild::set_target_architecture_data() in build.o
      std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > std::__1::operator+<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> > const&, wchar_t const*) in build.o
      CEXEBuild::init_shellconstantvalues() in build.o
      ...
  "_iconv", referenced from:
      WCFromCodePoint(wchar_t*, unsigned int, unsigned int) in utf.o
      NStreamLineReader::ReadLine(wchar_t*, unsigned int) in utf.o
      WideCharToMultiByte(unsigned int, unsigned int, wchar_t const*, int, char*, int, char const*, int*) in util.o
      nsis_iconv_reallociconv(void*, char**, unsigned long*, char**, unsigned long&) in util.o
      MultiByteToWideChar(unsigned int, unsigned int, char const*, int, wchar_t*, int) in util.o
     (maybe you meant: nsis_iconv_get_host_endian_ucs4_code(), nsis_iconv_reallociconv(void*, char**, unsigned long*, char**, unsigned long&) )
  "_iconv_close", referenced from:
      StrSetUTF16LE(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >&, void const*) in utf.o
      WCFromCodePoint(wchar_t*, unsigned int, unsigned int) in utf.o
      iconvdescriptor::~iconvdescriptor() in utf.o
      DupWCFromBytes(void*, unsigned int, unsigned int) in utf.o
      WCToUTF16LEHlpr::Create(wchar_t const*, unsigned int) in utf.o
      NOStream::WriteString(wchar_t const*, unsigned long) in utf.o
      NStreamLineReader::ReadLine(wchar_t*, unsigned int) in utf.o
      ...
  "_iconv_open", referenced from:
      StrSetUTF16LE(std::__1::basic_string<wchar_t, std::__1::char_traits<wchar_t>, std::__1::allocator<wchar_t> >&, void const*) in utf.o
      CharEncConv::Initialize(unsigned int, unsigned int) in utf.o
      WCFromCodePoint(wchar_t*, unsigned int, unsigned int) in utf.o
      DupWCFromBytes(void*, unsigned int, unsigned int) in utf.o
      WCToUTF16LEHlpr::Create(wchar_t const*, unsigned int) in utf.o
      NOStream::WriteString(wchar_t const*, unsigned long) in utf.o
      NStreamLineReader::ReadLine(wchar_t*, unsigned int) in utf.o
      ...
ld: symbol(s) not found for architecture x86_64
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
scons: *** [build/urelease/makensis/makensis] Error 1
scons: building terminated because of errors.
builder for '/nix/store/2b0gk7l1m4zda6mxj7jk8754q9y2nm2p-nsis-3.06.1.drv' failed with exit code 2
error: build of '/nix/store/2b0gk7l1m4zda6mxj7jk8754q9y2nm2p-nsis-3.06.1.drv' failed
```

--- a/pkgs/development/tools/nsis/default.nix
+++ b/pkgs/development/tools/nsis/default.nix
@@ -68,7 +68,7 @@ stdenv.mkDerivation rec {
     description = "A free scriptable win32 installer/uninstaller system that doesn't suck and isn't huge";
     homepage = "https://nsis.sourceforge.io/";
     license = licenses.zlib;
-    platforms = platforms.linux ++ platforms.darwin;
+    platforms = platforms.unix;
     maintainers = with maintainers; [ pombeirp ];
   };
 }

--- a/pkgs/development/tools/nsis/default.nix
+++ b/pkgs/development/tools/nsis/default.nix
@@ -33,9 +33,14 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ sconsPackages.scons_3_1_2 ];
   buildInputs = [ zlib ] ++ lib.optionals stdenv.isDarwin [ libiconv ];
 
-  nsisIncludes = symlinkJoin {
-    name = "includes";
-    paths = [ zlib.dev ] ++ lib.optionals stdenv.isDarwin [ libiconv ];
+  CPPPATH = symlinkJoin {
+     name = "nsis-includes";
+     paths = [ zlib.dev ] ++ lib.optionals stdenv.isDarwin [ libiconv ];
+  };
+
+  LIBPATH = symlinkJoin {
+    name = "nsis-libs";
+    paths = [ zlib ] ++ lib.optionals stdenv.isDarwin [ libiconv ];
   };
 
   sconsFlags = [
@@ -43,13 +48,17 @@ stdenv.mkDerivation rec {
     "SKIPPLUGINS=all"
     "SKIPUTILS=all"
     "SKIPMISC=all"
-    "APPEND_CPPPATH=${nsisIncludes}/include"
-    "APPEND_LIBPATH=${zlib}/lib"
     "NSIS_CONFIG_CONST_DATA=no"
-  ];
+  ] ++ lib.optional stdenv.isDarwin "APPEND_LINKFLAGS=-liconv";
 
-  preBuild = lib.optionalString stdenv.isLinux ''
-    sconsFlagsArray+=("PATH=$PATH")
+  preBuild = ''
+    sconsFlagsArray+=(
+      "PATH=$PATH"
+      "CC=$CC"
+      "CXX=$CXX"
+      "APPEND_CPPPATH=$CPPPATH/include"
+      "APPEND_LIBPATH=$LIBPATH/lib"
+    )
   '';
 
   prefixKey = "PREFIX=";

